api/v1alpha2/linodecluster_webhook.go

```diff
@@ -97,6 +97,21 @@ func (r *LinodeCluster) validateLinodeClusterSpec(ctx context.Context, client Li
 		errs = append(errs, err)
 	}
 
+	if r.Spec.Network.LoadBalancerType == "dns" {
+		if r.Spec.Network.DNSRootDomain == "" {
+			var domainError field.Error
+			domainError.Field = "dnsRootDomain needs to be set when LoadBalancer Type is DNS"
+			domainError.Type = field.ErrorTypeRequired
+			errs = append(errs, &domainError)
```

Comment:
```suggestion
			errs = append(errs, &field.Error{
				Field: "dnsRootDomain needs to be set when LoadBalancer Type is DNS",
				Type: field.ErrorTypeRequired,
			})
```

Reply:
good call ... implemented in both locations